### PR TITLE
DAW-007 | Crear Pag Perfil de Usuario

### PR DESCRIPTION
**Pull request by Davejim-30**
Codigo del perfil base casi listo. Por favor probar desde cs code personal

### CODE REVIEW

File: src/pages/SignUp.js [hunk 1]
```diff
@@ -112,14 +111,14 @@ function SignUp() {
           <Button variant="contained" color="primary" onClick={handleSubmit}>
             Registrarse
           </Button>
-
         </Box>
         <Typography variant="body1" align="center">
           ¿Ya tienes una cuenta? <Link to="/signin">Iniciar sesión</Link>
         </Typography>
       </Box>
+      <Perfil nameProp={name} lastNameProp={lastName} emailProp={email} levelProp={level} />
```
Comment (itslarlee) [inline src/pages/SignUp.js:119]: El componente de perfil no deberia de verse en el sign up page

image

File: src/pages/Perfil.js [hunk 2]
```diff
@@ -1,9 +1,189 @@
...
+  width: '100%',
+  marginTop: (theme) => theme.spacing(1),
+});
+
+const SubmitButton = styled(Button)(({ theme }) => ({
+  margin: theme.spacing(3, 0, 2),
+}));
+
+function Perfil({ nameProp, lastNameProp, emailProp, levelProp }) {
+  const [name, setName] = useState(nameProp);
```
Comment (itslarlee) [inline src/pages/Perfil.js:28]: En vez de venir de los props, la informacion deberia venir de la base de datos
Yo cree un hook para que esto sea mas facil

https://github.com/itslarlee/DAW/blob/3aeb4f9f195f5e8773f6e87c0277c136a4f6063b/src/providers/IdentityProvider.js#L1-L29

Y aqui es un ejemplo de como se usa

https://github.com/itslarlee/DAW/blob/3aeb4f9f195f5e8773f6e87c0277c136a4f6063b/src/providers/RouteProvider.js#L15

**Review by itslarlee — CHANGES_REQUESTED** [review]

File: src/pages/Perfil.js (hunk 2 above)
Comment (Davejim-30) [reply src/pages/Perfil.js:28]: Aqui esta el codigo corregido

import { styled } from '@mui/material';
import { Avatar, Button, TextField, Typography } from '@mui/material';
import React, { useState, useEffect } from 'react';
import { useIdentity } from '../providers/IdentityProvider';
import { doc, getDoc } from 'firebase/firestore';
import { db } from '../firebase';

const Root = styled('div')({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  padding: (theme) => theme.spacing(4),
});

const AvatarStyled = styled(Avatar)({
  width: (theme) => theme.spacing(12),
  height: (theme) => theme.spacing(12),
  marginBottom: (theme) => theme.spacing(2),
});

const Form = styled('form')({
  width: '100%',
  marginTop: (theme) => theme.spacing(1),
});

const SubmitButton = styled(Button)(({ theme }) => ({
  margin: theme.spacing(3, 0, 2),
}));

function Perfil() {
  const { user } = useIdentity();
  const [name, setName] = useState('');
  const [lastName, setLastName] = useState('');
  const [email, setEmail] = useState('');
  const [level, setLevel] = useState('');
  const [isEditingName, setIsEditingName] = useState(false);
  const [isEditingLastName, setIsEditingLastName] = useState(false);
  const [isEditingEmail, setIsEditingEmail] = useState(false);

  useEffect(() => {
    const fetchUserData = async () => {
      if (user) {
        const docRef = doc(db, 'users', user.uid);
        const docSnap = await getDoc(docRef);
        if (docSnap.exists()) {
          const userData = docSnap.data();
          setName(userData.name || '');
          setLastName(userData.lastName || '');
          setEmail(userData.email || '');
          setLevel(userData.level || '');
        }
      }
    };

    fetchUserData();
  }, [user]);

  const handleNameChange = (event) => {
    if (isEditingName) {
      setName(event.target.value);
    }
  };

  const handleLastNameChange = (event) => {
    if (isEditingLastName) {
      setLastName(event.target.value);
    }
  };

  const handleEmailChange = (event) => {
    if (isEditingEmail) {
      setEmail(event.target.value);
    }
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    if (isEditingName || isEditingLastName || isEditingEmail) {
      console.log('Datos actualizados:', { name, lastName, email });
      setIsEditingName(false);
      setIsEditingLastName(false);
      setIsEditingEmail(false);
    }
  };

  const handleEditName = () => {
    setIsEditingName(true);
  };

  const handleEditLastName = () => {
    setIsEditingLastName(true);
  };

  const handleEditEmail = () => {
    setIsEditingEmail(true);
  };

  return (
    <Root>
      <Typography component="h1" variant="h5">
        Perfil
      </Typography>
      <AvatarStyled />
      <Form onSubmit={handleSubmit} noValidate>
        <TextField
          variant="outlined"
          margin="normal"
          required
          fullWidth
          id="name"
          label="Nombre"
          name="name"
          autoComplete="given-name"
          autoFocus
          value={name}
          onChange={handleNameChange}
          InputProps={{
            readOnly: !isEditingName,
          }}
        />
        {!isEditingName && (
          <Button variant="outlined" onClick={handleEditName}>
            Editar
          </Button>
        )}

        <TextField
          variant="outlined"
          margin="normal"
          required
          fullWidth
          id="lastName"
          label="Apellido"
          name="lastName"
          autoComplete="family-name"
          value={lastName}
          onChange={handleLastNameChange}
          InputProps={{
            readOnly: !isEditingLastName,
          }}
        />
        {!isEditingLastName && (
          <Button variant="outlined" onClick={handleEditLastName}>
            Editar
          </Button>
        )}

        <TextField
          variant="outlined"
          margin="normal"
          required
          fullWidth
          id="email"
          label="Correo electrónico"
          name="email"
          autoComplete="email"
          value={email}
          onChange={handleEmailChange}
          InputProps={{
            readOnly: !isEditingEmail,
          }}
        />
        {!isEditingEmail && (
          <Button variant="outlined" onClick={handleEditEmail}>
            Editar
          </Button>
        )}

        <SubmitButton
          type="submit"
          fullWidth
          variant="contained"
          color="primary"
          disabled={!isEditingName && !isEditingLastName && !isEditingEmail}
        >
          Guardar cambios
        </SubmitButton>
      </Form>
    </Root>
  );
}

export default Perfil;

File: src/pages/SignUp.js (hunk 1 above)
Comment (itslarlee) [reply src/pages/SignUp.js:146]: @Davejim-30, creo que puedes eliminar esta linea

File: src/pages/Perfil.js [hunk 3]
```diff
@@ -1,9 +1,200 @@
...
+          variant="contained"
+          color="primary"
+          disabled={!isEditingName && !isEditingLastName && !isEditingEmail}
+        >
+          Guardar cambios
+        </SubmitButton>
+
+        {isSaved && (
+          <Typography variant="body1" color="success">
+            Los cambios han sido guardados correctamente.
```
Comment (itslarlee) [inline src/pages/Perfil.js:192]: Si puedes en vez de ponerlo en texto puedes usar un alert
https://mui.com/material-ui/react-alert/

File: src/pages/Perfil.js [hunk 4]
```diff
@@ -1,9 +1,200 @@
...
+          </Typography>
+        )}
+      </Form>
+    </Root>
+  );
```
Comment (itslarlee) [inline src/pages/Perfil.js:197]: Aun falta que el usuario pueda ver, agregar, editar y eliminar los registros de mediciones
Lo del peso, grasa, etc

**Review by itslarlee — CHANGES_REQUESTED** [review]

**Review by itslarlee — APPROVED** [review]
Se ve muy bien David! Me gustó mucho la tabla de las mediciones!

image

**Pull request merged**